### PR TITLE
feat: Refactor to integration testing setup (SC-567)

### PR DESCRIPTION
**Pull request by lucas-manuel**

### CODE REVIEW

File: test/fork/ForkTestBase.t.sol [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+
+contract ForkTestBase is DssTest {
+
+    /**********************************************************************************************/
+    /*** Mainnet addresses                                                                      ***/
+    /**********************************************************************************************/
+
+    bytes32 constant ilk = "ILK-A";
+
+    uint256 constant INK = 1e12 * 1e18;  // Ink initialization amount
```
[the remark this answers is not in the document]
Comment (lucas-manuel) [reply test/fork/ForkTestBase.t.sol:54]: Its used in the inheriting VaultCalls contract assertions

File: test/fork/ForkTestBase.t.sol [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+
+    bytes32 constant DEFAULT_ADMIN_ROLE = 0x00;
+
+    bytes32 constant CONTROLLER = keccak256("CONTROLLER");
+    bytes32 constant FREEZER    = keccak256("FREEZER");
+    bytes32 constant RELAYER    = keccak256("RELAYER");
```
[the remark this answers is not in the document]
Comment (lucas-manuel) [reply test/fork/ForkTestBase.t.sol:60]: Updated

File: test/unit/UnitTestBase.t.sol [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+            address(snst)
+        );
+
+        // Done with spell by pause proxy
+        vm.startPrank(admin);
+
+        ethereumController.grantRole(FREEZER, freezer);
+        ethereumController.grantRole(RELAYER, relayer);
+
+        almProxy.grantRole(FREEZER,    freezer);
```
Comment (hexonaut) [inline test/unit/UnitTestBase.t.sol:55]: ALM Proxy does not have a freezer role.

**Review by hexonaut — CHANGES_REQUESTED** [review]
A few issues. `MockJug` is unused. Also, I noticed `ALMProxy` is missing a way to send ETH to it. It should have a `receive()` special function.

Not sure why CI is failing, but tests work locally.

File: src/EthereumController.sol [hunk 4]
```diff
@@ -156,12 +169,24 @@ contract EthereumController is AccessControl {
     /**********************************************************************************************/
 
     function swapNSTToUSDC(uint256 usdcAmount) external onlyRole(RELAYER) isActive {
-        uint256 conversionFactor = psm.to18ConversionFactor();
+        uint256 wadAmount = usdcAmount * psm.to18ConversionFactor();
```
[the remark this answers is not in the document]
Comment (lucas-manuel) [reply src/EthereumController.sol:172]: Updated

File: src/EthereumController.sol [hunk 5]
```diff
@@ -172,17 +197,31 @@ contract EthereumController is AccessControl {
     }
 
     function swapUSDCToNST(uint256 usdcAmount) external onlyRole(RELAYER) isActive {
+        uint256 wadAmount = usdcAmount * psm.to18ConversionFactor();
```
[the remark this answers is not in the document]
Comment (lucas-manuel) [reply src/EthereumController.sol:200]: Updated

File: test/fork/ForkTestBase.t.sol [hunk 6]
```diff
@@ -0,0 +1,233 @@
...
+
+        SNstConfig memory snstConfig = SNstConfig({
+            nstJoin: address(nstInst.nstJoin),
+            nst: address(nstInst.nst),
+            nsr: 1000000001547125957863212448
+        });
+
+        AllocatorIlkConfig memory ilkConfig = AllocatorIlkConfig({
+            ilk            : ilk,
+            duty           : 1000000001243680656318820312,
```
[the remark this answers is not in the document]
Comment (lucas-manuel) [reply test/fork/ForkTestBase.t.sol:162]: Updated

File: test/fork/ForkTestBase.t.sol [hunk 7]
```diff
@@ -0,0 +1,233 @@
...
+            maxLine        : 100_000_000 * RAD,
+            gap            : 10_000_000 * RAD,
+            ttl            : 1 days,
```
[the remark this answers is not in the document]
Comment (lucas-manuel) [reply test/fork/ForkTestBase.t.sol:165]: Updated

File: test/fork/ForkTestBase.t.sol [hunk 8]
```diff
@@ -0,0 +1,233 @@
...
+        /*** Step 4: Configure ALM system in allocation system ***/
+
+        vm.startPrank(SPARK_PROXY);
+
+        IVaultLike(ilkInst.vault).rely(address(almProxy));
+
+        ethereumController.grantRole(FREEZER, freezer);
+        ethereumController.grantRole(RELAYER, relayer);
+
+        almProxy.grantRole(FREEZER,    freezer);
```
[the remark this answers is not in the document]
Comment (lucas-manuel) [reply test/fork/ForkTestBase.t.sol:206]: Fixed

File: test/unit/UnitTestBase.t.sol (hunk 3 above)
Comment (lucas-manuel) [reply test/unit/UnitTestBase.t.sol:55]: Fixed

**Review by hexonaut — APPROVED** [review]

**Pull request merged**